reference/src/optimizations/return_value_optimization.md

```diff
@@ -0,0 +1,20 @@
+We should turn
+
+```rust
+// y unused
+let mut x = f();
+g(&mut x);
+y = x;
+// x unused
+```
+
+into
+
+```rust
+y = f();
+g(&mut y);
+```
+
+to avoid a copy.
+
+The potential issue here is `g` storing the pointer it got as an argument elsewhere.
```

Comment:
If `x` is unused after the call to `g()`, can that pointer be used for anything ?

Reply:
`g` could have stored it in a global variable and enable other code to use it, for example.